tests/e2e/utils/DriverHelper.ts

```diff
@@ -194,99 +202,121 @@ export class DriverHelper {
-        attribute: string,
-        visibilityTimeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT): Promise<string> {
-
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
+    public async waitAndGetElementAttribute(elementLocator: By, attribute: string,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT): Promise<string> {
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitAndGetElementAttribute ${elementLocator} attribute: '${attribute}'`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, visibilityTimeout);
-
             try {
-                const attributeValue = await element.getAttribute(attribute);
-                return attributeValue;
-            } catch (err) {
-                if (err instanceof error.StaleElementReferenceError) {
-                    await this.wait(polling);
+                const element: WebElement = await this.waitVisibility(elementLocator, polling);
+
+                try {
+                    const attributeValue = await element.getAttribute(attribute);
+                    return attributeValue;
+                } catch (err) {
+                    if (err instanceof error.StaleElementReferenceError) {
+                        await this.wait(polling);
+                        continue;
+                    }
+
+                    throw err;
+                }
+            } catch(err) {
+                if (i >= attempts) {
+                    Logger.error(`DriverHelper.waitAndGetElementAttribute - failed with exception, out of attempts - ${err}`);
+                    throw err;
+                } else {
                     continue;
                 }
-
-                throw err;
             }
         }
 
         throw new error.TimeoutError(`Exceeded maximum gettin of the '${attribute}' attribute attempts, from the '${elementLocator}' element`);
     }
 
-    public async waitAndGetCssValue(elementLocator: By,
-        cssAttribute: string,
-        visibilityTimeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT): Promise<string> {
-
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
+    public async waitAndGetCssValue(elementLocator: By, cssAttribute: string,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT): Promise<string> {
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitAndGetCssValue ${elementLocator} cssAttribute: ${cssAttribute}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, visibilityTimeout);
-
             try {
```

Comment:
2 levels of try/catch

Reply:
I had issues for the methods failing after times that did not match the timeout without a sensual exception message.
The waitVisibility method was also failing with ElementNotFound that was never caught and caused issue with the logic of the tests.
The second layer of try-catch is there to prevent the method failing without reaching the desired amount of attempts/timeout and if the issue persists, it throws the correct exception.